### ISSUE TITLE
OpenToonz discards dpi information when editing bitmap image (PNG)

### ISSUE THREAD

**Issue by morevnaproject** [opentoonz/opentoonz#1935]
**Steps to reproduce:**

1. Launch OpenToonz. Create new scene. Import this file into OpenToonz (just a PNG file with resolution set to 300dpi) - https://download.tuxfamily.org/morevna/blog/2018/04/seashel.png

![screenshot_004](https://user-images.githubusercontent.com/332868/39353937-4f38846e-4a33-11e8-9e52-b3665f555ffe.png)

You can right-click on image cell and choose "info.." to see that dpi information is there - 

2. Use selection tool to select a piece of image. Hit Ctrl+X to cut that piece from image.
![screenshot_005](https://user-images.githubusercontent.com/332868/39354085-b5ad054e-4a33-11e8-86fc-90fded1e8960.png)

3. Hit Ctrl+S to save, Quit  OpenToonz.

4. Launch OpenToonz again and open the same scene. You will see the image now displayed bigger:
![screenshot_006](https://user-images.githubusercontent.com/332868/39354093-bad45090-4a33-11e8-89c7-44b9928accb3.png)

...right-click on image cell and choose "info.." to see that dpi information is lost (this is what causing image do display bigger):
![screenshot_008](https://user-images.githubusercontent.com/332868/39354583-09a01c44-4a35-11e8-93a3-e155adb44d33.png)

**Comment by RodneyBaker** [comment]
At some point we really need to address the DPI issue in OpenToonz as DPI is a problematic term.
This is further complicated where OpenToonz has a Camera "DPI" which can be even more confusing.

But that camera DPI (or resolution) actually addresses your issue in that if we Right Click on the level in the Xsheet there is a button that states "Use Camera DPI".  Selecting that for both your Levels will set them to same size.

Now, to the thought of the underlying problem you specify in this report...
If OpenToonz is actually factoring in the DPI (print resolution) of the image to display the image then that may need to be reconsidered as thee is no direct correlation between dots per inch placed on a piece of paper and the pixels placed on the screen.  A change of DPI setting in an image (or lack of a DPI setting entirely) should not make any difference in what we see on our screen.

**Comment by shun-iwasawa** [comment]
Yes, I once checked and confirmed that OT has an ability to load dpi info from PNG but doesn't save it.
( That's why we cannot add PNG format as an option for default raster level. Please see my [previous comment](https://github.com/opentoonz/opentoonz/pull/1877#issuecomment-375264333). )
At first glance OT seems to obtain dpi information on saving PNG [here](https://github.com/opentoonz/opentoonz/blob/master/toonz/sources/image/png/tiio_png.cpp#L808-L809) but I'm not sure it works. 

@RodneyBaker 
Considering DPI enables OT to have virtual camera stand in which the images are placed in physical scale. And IMO it is somewhat easier-to-grasp than work on pixels especially for professionals in Japanese animation industry, since they usually give another staff instructions of camera work with physical scale (millimeter per frame) just like conventional animation production workflow using actual camera stand and film.

**Comment by DizzyDreamerZ** [comment]
I've been experiencing the same problem. For the sake of consistency in the project I'm currently working on, I'm using a program called DesignDoll to make reference images for the keyframes which I import into OT and draw along them to keep the anatomy and everything....well, consistent. Once every now and again, when I open OT, one image from the level with the reference image sequence gets enlarged. More often than not, it's the image sitting in the fame I last worked on before closing OT in the last session.
Here's an example.
![1](https://user-images.githubusercontent.com/40675289/42063549-e8ca74a6-7b3a-11e8-98ce-9fc8fbf1d419.PNG)
This is how it's normally supposed to be, with me drawing along the positioned models from DesignDoll. The last frame I had workd on as of when I took these screenshots was 141. And, this is how I found said frame when I opened OT today.
![2](https://user-images.githubusercontent.com/40675289/42063634-3e7bd066-7b3b-11e8-8254-7b27763ca930.PNG)
Keep in mind that this is not the first time this has happened so I've been trying to figure out how to resolve the issue. A method I found out worked for me (though probably not a proper fix) is to select the entire image in that particular frame with the Selection Tool after which I decrease its overall size to exactly 55.5%. For some reason, whenever an image blows up like this, it's always by the exact same amount, so decreasing it exactly to 55.5% seems to return it to how it should be.

**Comment by RodneyBaker** [comment]
This is to add the documented workaround to various DPI/scaling issues:

Some image formats do not store DPI information and some programs may discard DPI data upon saving.  When using images created in programs other than Opentoonz it is good to consider the source image format and its ability to store DPI.
Workaround:  Use an image format that will store DPI information.

_Table Listing of image formats and the ability to support for DPI data to be added here._

Additional Considerations:
Opentoonz internal formats will store DPI information but image formats exported which may be reimported later into Opentoonz may no longer contain that information due to the format itself or loss of data while being processed in the other program(s).

**Comment by RodneyBaker** [comment]
The DPI issue is a problem but also one that doesn't have a clear path forward to a full solution.
Of options currently available the most likely solution would be to utilize a file format that stores DPI information and make sure images ingested into Opentoonz contain that information.
A close second might be to always maintain the default DPI settings inside Opentoonz at the desired value. and to make sure all overrides are set to on so that ingested images default to the desired DPI.

Perhaps a path forward would be to collect all known settings for DPI (via Preferences and other locations) into a tab in Preferences where the user can more easily find and set values and options.

Opentoonz could also be made to look for a sidecar file containing DPI information and (if present) use that to set or override the image/image sequence resolution.

Regardless of the road ahead it appears that users need to have a clear view of what DPI they are using as well as how Opentoonz will treat images as they are brought into the program.  

An additional consideration must be made for resources adjusted for use in Opentoonz but not changed in the original as editing via external applications can alter the workflow even more.

**Comment by RattusaurusRex** [comment]
It seems like Pixels Only mode and/or 'Use Camera DPI for All Imported Images' in the Preferences will prevent the imported images from changing size.  

![set](https://user-images.githubusercontent.com/60277069/90995388-a8d74700-e570-11ea-8aaf-2880b67b4185.png)

If you don't have either of these turned on, another method to "fix" your images afterwards seems to be to manually match your level dpi to the camera dpi. Someone on DarrenT's Discord server was having this problem, in which 3 of his levels had lost their DPI data.

His scene: [Odog.zip](https://github.com/opentoonz/opentoonz/files/5115387/Odog.zip)

Video of how to match the camera and level dpi to fix the scene: [dpi.zip](https://github.com/opentoonz/opentoonz/files/5115394/dpi.zip)

**Comment by openingapps** [comment]
The problem you describe is real.  Opentoonz pretty much ignores dpi thoughout the code base.  BUT there is a dpi conversion that happens when images are first read.

The problem seems to be in a bug when opentoonz saves files.  In a good scene file (.tnz) file, you will see:
            <info dpix="120.000000" dpiy="120.000000"/>
If this gets lost somehow, the next time you open the file, the image is scaled up at read time.  The amount of scale-up seems to be related to the amount of transparent pixels you have bordering your image.  The more transparent border, the more scaling.

Additionally, you sometimes see this added to the tnz file:
       <info dpiType="image"/>
When this is in the file, you always get bad behavior!

I have not been able to reliably reproduce the problem, but my guess is that images are accidentally being changed from fixed scaling to this weird "dpiType="image" scaling.  This does not hurt you right away.  You only see it when you load the file the next time.
